pkg/infrabin/server.go

```diff
@@ -0,0 +1,71 @@
+package infrabin
+
+import (
+	"log"
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+type HTTPServer struct {
+	Name string
+	Server *http.Server
+}
+
+
+func (s *HTTPServer) ListenAndServe() {
+	log.Printf("Starting %s server on %s", s.Name, s.Server.Addr)
+	if err := s.Server.ListenAndServe(); err != http.ErrServerClosed {
+		log.Fatal("HTTP server crashed", err)
+	}
+}
+
+
+func (s *HTTPServer) Shutdown() {
+	log.Printf("Shutting down %s server with 15s graceful shutdown", s.Name)
+	ctx, cancel := context.WithTimeout(context.Background(), 15 * time.Second)
```

Comment:
Will wait up to 15 seconds or exit immediately
Followed this https://github.com/gorilla/mux#graceful-shutdown

Reply:
Awesome, thanks